apps/wing/src/wingc.ts

```diff
@@ -0,0 +1,117 @@
+import debug from "debug";
+import { readFile } from "fs/promises";
+import { resolve } from "path";
+import { WASI } from "wasi";
+
+const log = debug("wing:compile");
+
+const WINGSDK_RESOLVED_PATH = require.resolve("@winglang/sdk");
+const WINGSDK_MANIFEST_ROOT = resolve(WINGSDK_RESOLVED_PATH, "../..");
+const WINGC_WASM_PATH = resolve(__dirname, "../wingc.wasm");
+
+export type WingCFunction =
+  | "wingc_compile"
+  | "wingc_on_did_open_text_document"
+  | "wingc_on_did_change_text_document"
+  | "wingc_on_completion"
+  | "wingc_on_semantic_tokens";
+
+// When WASM stuff returns a value, we need both a pointer and a length,
+// We are using 32 bits for each, so we can combine them into a single 64 bit value.
+// This is a bit mask to extract the low order 32 bits.
+// https://stackoverflow.com/questions/5971645/extracting-high-and-low-order-bytes-of-a-64-bit-integer
+const LOW_MASK = 2n ** 32n - 1n;
+const HIGH_MASK = BigInt(32);
+
+export interface WingCLoadOptions {
+  imports?: Record<string, any>;
+  preopens?: Record<string, string>;
+  env?: Record<string, string>;
+}
+
+export async function loadWingc(options: WingCLoadOptions) {
```

Comment:
We are already in the context of "wingc.ts":

```suggestion
export async function load(options: WingCompilerLoadOptions) {
```